src/pages/index.js

```diff
@@ -5,77 +5,210 @@ import PopupWithImage from "../components/PopupWithImage.js";
+// };
 
 //SELECTORS----------------------
 const cardsWrap = document.querySelector(".elements__cards");
 const profileEditButton = document.querySelector("#profile-edit-button");
 const addNewCardButton = document.querySelector(".profile__add-button");
 const addCardModal = document.querySelector("#add-card-modal");
 const addCardForm = addCardModal.querySelector(".modal__form");
-const addCardTitleInput = addCardForm.querySelector("#title-modal-input");
-const addCardUrlInput = addCardForm.querySelector("#image-url-modal-input");
+
+const editProfilePictureModal = document.querySelector(
+  "#profile-picture-modal"
+);
+const editProfilePictureForm =
+  editProfilePictureModal.querySelector(".modal__form");
 const profileEditModal = document.querySelector("#profile-edit-modal");
-const profileName = document.querySelector(".profile__name");
-const profileDescription = document.querySelector(".profile__description");
+const profileEditForm = profileEditModal.querySelector(".modal__form");
 const profileNameInput = document.querySelector("#name-modal-input");
 const profileDescriptionInput = document.querySelector(
   "#description-modal-input"
 );
-const profileEditForm = profileEditModal.querySelector(".modal__form");
-const previewImageModalWindow = document.querySelector("#modal-preview");
+const profileImage = document.querySelector(".profile__image");
+const profileImageButton = document.querySelector(".profile__icon");
+const avatarModal = document.querySelector("#profile-picture-modal");
+const avatarModalButton = avatarModal.querySelector(".modal__button");
+// const deleteButton = document.querySelector(".elements__trash-button");
 
 //Section Render
 
-const cardsSection = new Section(
+// const cardsSection = new Section(
+//   {
+//     renderer: renderCard,
+//   },
+//   ".elements__cards"
+// );
+
+//User Info Render
+
+const userInfo = new UserInfo(
+  ".profile__name",
+  ".profile__description",
+  ".profile__image"
+);
+
+//API Render
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "7b855de8-b589-492b-9e1c-b631d0b0f4c2",
+    "Content-Type": "application/json",
+  },
+});
+
+const cardList = new Section(
   {
-    items: initialCards,
-    renderer: renderCard,
+    renderer: (cardData) => {
+      cardList.addItem(createCard(cardData));
+    },
   },
   ".elements__cards"
 );
 
+api
+  .getInitialCards()
+  .then((res) => {
+    // console.log(">>Initial Cards", cardData);
+    cardList.renderItems(res.reverse());
+    console.log(res);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo(userData);
+    userInfo.setUserAvatar(userData.avatar);
+  })
+  .catch((err) => {
+    console.error(err);
+  });
+
 //FUNCTIONS----------------------
 
+function setButtonText(button, text) {
```

Comment:
Needs correction: you don't need this function, since you have the logic for rendering loading in your popup classes